misc-utils/getopt.c

```diff
@@ -126,53 +129,40 @@ static void print_normalized(const struct getopt_control *ctl, const char *arg)
 	 * and an opening quote! We need also the global opening and closing
 	 * quote, and one extra character for '\0'.
 	 */
-	buf = xmalloc(strlen(arg) * 4 + 3);
-	bufptr = buf;
+	buf = xmalloc(strlen(arg) * 4 + 1);
 
-	for (*bufptr++ = '\''; *argptr; argptr++) {
+	for (bufptr = buf; *argptr; argptr++) {
 		if (ctl->shell == TCSH) {
 			switch (*argptr) {
 			case '\\':
-				/* Backslash: replace it with: '\\' */
-				*bufptr++ = '\\';
-				*bufptr++ = '\\';
+				/* Backslash: replace it with: \\ */
+				bufptr += sprintf(bufptr, BACKSLASH BACKSLASH);
 				continue;
```

Comment:
Frankly, snprint() is a monster;  the orignal *bufptr++ is more straightforward solution if you do not need to format any string.